assets/js/stripe.js

```diff
@@ -791,31 +816,38 @@ jQuery( function( $ ) {
 		/**
 		 * Queries the server for a PaymentIntent.
 		 *
+		 * @param {string} savedCard The ID of the saved card. (Optional)
 		 * @return {Promise} A promise that resolves to a payment intent.
 		 */
-		getIntent: function() {
-			return new Promise( function( resolve, reject ) {
-				if ( wc_stripe_form.paymentIntent ) {
-					return resolve( wc_stripe_form.paymentIntent );
-				}
+		getIntent: function( savedCard ) {
+			savedCard = savedCard || 'new';
+
+			// If we have an intent for this card already, use it.
+			if ( wc_stripe_form.paymentIntentCache.hasOwnProperty( savedCard ) ) {
+				return Promise.resolve( wc_stripe_form.paymentIntentCache[ savedCard ] );
+			}
 
+			return new Promise( function( resolve ) {
 				$.ajax( {
 					url: wc_stripe_form.getAjaxURL( 'create_intent' ),
 					type: 'post',
+					data: {
+						saved_card: ( 'new' === savedCard ? null : savedCard ),
+					},
 					dataType: 'json',
 					success: function( result ) {
 						if ( result.error ) {
-							reject( result.error );
+							return $( document.body ).trigger( 'stripeError', { error: result.error } );
```

Comment:
This is trippy. You're basically stopping the Promise from ever resolving or rejecting if there's an error. That works, but it's quite hard to understand. Please put comments somewhere indicating this, or keep the explicit error handling where this function is called.

Reply:
Explicit error handling kept 👍